### PR TITLE
[KTO] fix tokenization bugs

### PR DESCRIPTION
**Pull request by kawine**
- fixed the tokenization bugs mentioned here https://github.com/huggingface/trl/issues/1401
- made some minor changes to the doc to clarify this issue https://github.com/huggingface/trl/issues/1387

cc @kashif

### CODE REVIEW

**Review by lewtun — COMMENTED** [review]
Thanks a lot for the fix @kawine ! Would you mind adding a small unit test for the `tokenize_row()` method to ensure we don't regress on this in future 🙏 ?

**Comment by kashif** [comment]
@kawine you can use this test:

```python
    def test_kto_trainer_tokenize_row(self):
        with tempfile.TemporaryDirectory() as tmp_dir:
            training_args = KTOConfig(
                output_dir=tmp_dir,
                per_device_train_batch_size=2,
                max_steps=3,
                remove_unused_columns=False,
                gradient_accumulation_steps=1,
                learning_rate=9e-1,
                evaluation_strategy="steps",
                beta=0.1,
            )

            dummy_dataset = self._init_dummy_dataset()

            trainer = KTOTrainer(
                model=self.model,
                ref_model=self.ref_model,
                args=training_args,
                tokenizer=self.tokenizer,
                train_dataset=dummy_dataset,
                eval_dataset=dummy_dataset,
            )

            row = dummy_dataset[0]

            # test that the row can be tokenized
            tokenized_row = trainer.tokenize_row(row)

            # Assert bos_token_id
            assert tokenized_row["prompt_input_ids"][0] == self.tokenizer.bos_token_id
            assert tokenized_row["completion_input_ids"][0] == self.tokenizer.bos_token_id
  ```

**Comment by kawine** [comment]
thanks! test has been added @kashif

File: tests/test_kto_trainer.py [hunk 1]
```diff
@@ -15,11 +15,11 @@
 import unittest
 
 import torch
+import wandb
 from datasets import Dataset
 from parameterized import parameterized
 from pytest import mark
 from transformers import AutoModelForCausalLM, AutoModelForSeq2SeqLM, AutoTokenizer
-
 from trl import KTOConfig, KTOTrainer
```
Comment (kashif) [inline tests/test_kto_trainer.py:23]:
```suggestion

from trl import KTOConfig, KTOTrainer
```

**Review by kashif — APPROVED** [review]

**Pull request merged**